src/federated.c

```diff
@@ -19,21 +19,22 @@ void FederatedConnectionBundle_connect_to_peers(FederatedConnectionBundle **bund
   }
 
   bool all_connected = false;
-  interval_t wait_before_retry = FOREVER; // Intialize to maximum so we can find the lowest requested.
+  interval_t wait_before_retry = FOREVER; // Initialize to maximum so we can find the lowest requested.
   while (!all_connected) {
     all_connected = true;
     for (size_t i = 0; i < bundles_size; i++) {
       FederatedConnectionBundle *bundle = bundles[i];
       NetworkChannel *chan = bundle->net_channel;
-      NetworkChannelState state = chan->get_connection_state(chan);
-      if (state != NETWORK_CHANNEL_STATE_CONNECTED) {
-        ret = chan->try_connect(chan);
-        switch (ret) {
-        case LF_OK:
-          bundle->network_channel_state_changed(bundle);
+      ret = chan->try_connect(chan);
+      if (ret == LF_OK) {
```

Comment:
Shouldn't also the channel state be updated if try_connect failes?

Reply:
Actually I am not sure because the channel itself should have already notified the bundle, I would need to look deeper into the runtime code tomorrow to see if that is covered